### PR TITLE
bugfix: workaround for 200 local variables limitation in luajit

### LINKED ISSUE #40: about 200 local variables limitation
Hello, @spacewander ,

The problem of 200 local variables limitation often occur to us when the json schema file geting bigger.
error info like:
```
main function has more than 200 local variables
```
I found 2 relative record from other repo:
1. https://github.com/Tieske/lua-resty-ljsonschema/pull/2
2. https://github.com/jdesgats/ljsonschema/issues/10

can we do this workaround too? thanks.

### PR DESCRIPTION
**Pull request by RocFang**
this pr tries to avoid 200 local variables limitation in luajit. 

@membphis would you like to have a review? thx:)

fix #40

### CODE REVIEW

**Comment by RocFang** [comment]
this pr dit not rewrite localvar or change every localvar to the new localvartab,
because local variables in for loops are a little special. 
But it should have coverd most cases.

File: lib/jsonschema.lua [hunk 1]
```diff
@@ -120,6 +120,19 @@ function codectx_mt:localvar(init, nres)
   return unpack(names)
 end
 
+function codectx_mt:localvartab(init, nres)
+    local names = {}
+    local nloc = self._nloc
+    nres = nres or 1
+    for i=1, nres do
```
Comment (membphis) [inline lib/jsonschema.lua:127]: style: `for i = 1, nres do`

File: lib/jsonschema.lua [hunk 2]
```diff
@@ -120,6 +120,19 @@ function codectx_mt:localvar(init, nres)
...
+      names[i] = sformat('locals.var_%d_%d', self._idx, nloc+i)
```
Comment (membphis) [inline lib/jsonschema.lua:128]: style: `nloc+i` -> `nloc + i`

**Review by membphis — COMMENTED** [review]
LGTM, except minor code style things

File: lib/jsonschema.lua (hunk 2 above)
Comment (RocFang) [reply lib/jsonschema.lua]: @membphis fixed the style problems.
And the same style problems in old localvar are also fixed:)

**Review by membphis — APPROVED** [review]

**Comment by membphis** [comment]
@RocFang many thx, merged right now

**Pull request merged**